app/src/main/java/com/mr/nemo/dragonfly/ui/screen/onboarding/OnboardingPage.kt

```diff
@@ -0,0 +1,170 @@
+    title: String,
+    description: String,
+    image: Painter,
+    isSwipeble: Boolean,
+    pageCount: Int,
+    currentPage: Int,
+    onNextClicked: () -> Unit,
+    onBackClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val typography = DragonFlyTheme.typography
+    val colors = DragonFlyTheme.colors
+    val spacing = DragonFlyTheme.spacing
+
+    Column(
+        modifier = modifier.fillMaxSize(),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        Spacer(modifier = Modifier.height(spacing.xLarge))
+
+        Image(
+            painter = image,
+            contentDescription = null,
+            modifier = Modifier
+                .weight(0.8f)
+                .fillMaxWidth(),
+            contentScale = ContentScale.FillHeight
+        )
+
+        Spacer(modifier = Modifier.height(spacing.large))
+
+        Column(
+            modifier = Modifier
+                .padding(horizontal = spacing.medium),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+
+            TitleText(
+                text = title,
+                color = colors.neutral2,
+                textAlign = TextAlign.Center
+            )
+
+            Spacer(modifier = Modifier.height(spacing.medium))
+
+            Text(
+                text = description,
+                style = typography.text2.regular,
+                color = colors.neutral4,
+                textAlign = TextAlign.Center
+            )
+
+            AnimatedVisibility(visible = isSwipeble) {
+                Column(
+                    verticalArrangement = Arrangement.Center,
+                    horizontalAlignment = Alignment.CenterHorizontally
+                ) {
+                    Spacer(modifier = Modifier.height(spacing.xLarge))
+
+                    Text(
+                        text = stringResource(R.string.swipe_for_more),
+                        style = typography.text1.medium,
+                        color = colors.neutral2,
+                    )
+                    Spacer(modifier = Modifier.height(spacing.xSmall))
+                    Icon(
+                        imageVector = Icons.Default.KeyboardArrowDown,
+                        contentDescription = stringResource(R.string.content_description_swipe_down_for_more),
+                        tint = colors.primary.main
+                    )
+                }
+            }
+
+            Spacer(modifier = Modifier.height(spacing.xLarge))
+
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                val buttonText by remember {
+                    derivedStateOf {
+                        if (currentPage == pageCount) {
+                            R.string.button_get_started
+                        } else {
+                            R.string.button_next
+                        }
+                    }
+                }
```

Comment:
@bohdanzdaniekin same here. Could you please share more info about this implementation?